lib/_stream_writable.js

```diff
@@ -642,15 +623,25 @@ Writable.prototype.end = function(chunk, encoding, cb) {
     this.uncork();
   }
 
+  if (typeof cb !== 'function')
+    cb = nop;
+
   // Ignore unnecessary end() calls.
-  if (!state.ending) {
+  // TODO(ronag): Compat. Allow end() after destroy().
+  if (!state.errored && !state.ending) {
```

Comment:
So now if someone calls `.end()` after an error stream won't have `state.ended` \ `state.ending` set up and therefore `stream.writableEnded` and explicit through `state` checks will remain false IIUC (due to `!state.errored` check here), is that ok?

Reply:
> is that ok?

I think it should be ok since the stream is `.destroyed` here. In my opinion it shouldn't be able to be `ended` ever once `.destroyed` however that's more breaking, `.errored` is a compromise.

i.e. how can a `end()` succeed when the stream has already failed?

Reply:
That's true, didn't think about the `destroy`. Could you please add a comment about this in the docs? (probably in `_write` error case, `'error'` event and in the `.end()` itself IIUC).

Reply:
I would prefer not to add anything about this in the docs until the TODO's are resolved and/or we are consistent. The behaviour here is already undocumented and inconsistent.

I don't think this change will affect anyone in practice.

Reply:
Makes sense since we might change this behavior though how quickly do you think we would be able to resolve these? Since in the case of streams such inconsistencies may hang around for quite a lot of time and IMO they better be documented in that case.

Though I completely agree with `it shouldn't be able to be ended ever once .destroyed`.

Reply:
> Makes sense since we might change this behavior though how quickly do you think we would be able to resolve these?

No idea. I'm trying to get it done in some fashion as soon as possible. Realistically, maybe sometime next year?

To be honest part of my disagreement is that I'm unsure how to and where to document this in order for it not to simply cause more confusion.